classes/base_decor.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0..9] # the word is cut off at 10 characters
+    else
+      @nameable.correct_name
+    end
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** Based on Ruby conventions, each class should be in its own file. I would recommend you move these classes into their respective files.